_posts/2022-11-20-v0.25-introducing-use-expressions.md

```diff
@@ -128,53 +128,136 @@ developer experience, and Gleam is still a small language compared to most.
 
 ## What does `use` do?
 
-lorem ipsum
-
-todo todo
-
+The `use` expression is a bit of syntactic sugar that turns all following
+expressions into an anonymous function that gets passed as an additional
+argument to a function call.
 
+For example, imagine I had a function called `with_file` which would open a
+file, pass the open file to a given function so it can read from or write to it,
+and then closes the file afterwards.
 
+```gleam
+// Define the function
+pub fn with_file(path, handler) {
+  let file = open(path)
+  handler(file)
+  close(file)
+}
 
+// Use it
+pub fn main() {
+  with_file("pokemon.txt", fn(file) {
+    write(file, "Oddish\n")
+    write(file, "Farfetch'd\n")
+  })
+}
+```
 
+With `use` this function could be called without extra indentation. This example
+below with `use` compiles to exactly the same code as the one above.
 
+```gleam
+pub fn main() {
+  use file <- with_file("pokemon.txt")
+  write(file, "Oddish\n")
+  write(file, "Farfetch'd\n")
+}
+```
 
+And it's not just limited to a single argument, functions of any arity can be
+used, including functions that accept no arguments at all.
 
+## What can it be used for?
 
+The `use` expression is highly generic and isn't restricted to any particular
+types, interfaces, or laws, so it can applied for many different things.
 
+It could be used to create HTTP middleware, as seen in the web service example above.
 
+```gleam
+pub fn require_method(request, method, continue) {
+  case request.method == method {
+    True -> continue()
+    False -> method_not_allowed()
+  }
+}
 
+pub fn handle_request(request) {
+  use <- require_method(request, Post)
+  // ...
+}
+```
 
+It could be used to replicate Go's [defer](https://gobyexample.com/defer)
+statements.
 
+```gleam
+pub fn defer(body, cleanup) {
```

Comment:
I think the arguments are backwards? This would print
```
Goodbye
Hello!
```

Reply:
Thank you